app/lib/repository/client.dart

```diff
@@ -11,3 +14,45 @@ Future<Client> login(String username, String password) async {
   Client client = await sdk.login(username, password);
   return client;
 }
+
+//Fetch user id from sdk
+Future<String> getUser(Future<Client> client) async {
+  Client _client = await client;
+  final String userId = await _client.userId();
+  return userId;
+}
```

Comment:
I've seen these a few times now and I don't understand them. You can just as well write `await (await client).user_id()` wherever you are currently calling `getUser`. What's the point of putting that into another function?

Reply:
The purpose was to just have more readability, as it would be nice to have all sdk functions seperately in a dart file so it doesn't get complicated to look out for bugs in the future. But I think thats why ffi file serves that case which I couldn't understood it earlier. I'll update it.